pkg/filesystem/path/trace.go

```diff
@@ -26,21 +26,50 @@ func (t *Trace) Append(component Component) *Trace {
 	}
 }
 
-func (t *Trace) writeToStringBuilder(sb *strings.Builder) {
+func (t *Trace) writeToUNIXStringBuilder(sb *strings.Builder) {
 	if t.parent != nil {
-		t.parent.writeToStringBuilder(sb)
+		t.parent.writeToUNIXStringBuilder(sb)
 		sb.WriteByte('/')
 	}
 	sb.WriteString(t.component.String())
 }
 
+func (t *Trace) writeToWindowsStringBuilder(sb *strings.Builder) error {
```

Comment:
Sorry for getting back to this, but giving it some more thought maybe the cleanest thing is to stick to a single `writeToStringBuilder()`, but extend it to take a `separator rune` argument.

With regards to validating the path for Windows, we can add an extra loop to `GetWindowsString()`:
```go
for tValidate := t; tValidate != nil; tValidate = tValidate.parent {
    if err := validateWindowsPathComponent(tValidate.component.String()); err != nil {
        return err
    }
}
```

Reply:
Sounds good!